### PR TITLE
Add scripts to aid PRs & releases

### PR DESCRIPTION
**Pull request by AndrewFerr**
This PR adds some scripts that do some maintenance-related tasks that I've been doing manually thus far, and are good to have down in writing:

- **tag-release** creates an appropriate tag for a new fork release, based on the release histories of the fork & upstream.
- **publish-to-npm** publishes the latest fork release branch to NPM & sets the correct package version while doing so.
- **prepare-patch-branch** creates a new branch that can be used to push commits that ought to be merged into both the fork & upstream.  The branch is based on the latest common ancestor commit of the fork & upstream, which makes it easier to merge into both repos.

### CODE REVIEW

File: scripts/prepare-patch-branch [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+for REMOTE in $(git remote); do
+    URL=$(git remote get-url $REMOTE)
+    if [[ $URL =~ "turt2live" ]]; then
+        UPSTREAM_REPO=$REMOTE
+    elif [[ $URL =~ "vector-im" ]]; then
+        FORK_REPO=$REMOTE
+    fi
+done
+
+function echoAndDo {
```
Comment (Half-Shot) [inline scripts/prepare-patch-branch:19]: Can this be done via https://stackoverflow.com/a/2853811?

File: scripts/publish-to-npm [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (Half-Shot) [inline scripts/publish-to-npm:1]: Sorry, I might be confused...Doesn't `yarn publish` by default pick the version from the package.json?

File: scripts/tag-release [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    while [[ -n $(git tag -l "v${THIS_VER}") ]]; do
+        THIS_VER=$(npx semver --preid ${PREID} -i prerelease $THIS_VER)
+    done
+    THIS_TAG="v${THIS_VER}"
+fi
+
+sed -i 's/\("version": "\).*\("\)/\1'$THIS_VER'\2/' package.json
+git add package.json
+git commit -m $THIS_TAG
+git tag -sm $THIS_TAG{,}
```
Comment (Half-Shot) [inline scripts/tag-release:47]: funky syntax alert :warning: 

what does `{,}` do?

**Review by Half-Shot — CHANGES_REQUESTED** [review]
Couple of things.

Generally these look like scripts scratching useful itches, so if they're working for you then we can merge them.

File: scripts/publish-to-npm (hunk 2 above)
Comment (AndrewFerr) [reply scripts/publish-to-npm:1]: You're right -- I must have put this in after having published while the version string was left as `develop`.

I also just discovered [the publishConfig setting for package.json](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#publishconfig), rendering the other purpose of this script obsolete.

File: scripts/prepare-patch-branch (hunk 1 above)
Comment (AndrewFerr) [reply scripts/prepare-patch-branch:19]: Not quite; the top suggestion of `set -v` prints every line, which effectively prints the whole script file.

What I went with is more like this other suggestion further down: https://stackoverflow.com/a/23342259

**Pull request merged**